extensions/wikia/Email/Controller/FounderDigestController.class.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+use Email\Check;
+use Email\Tracking\TrackingCategories;
+
+abstract class FounderDigestController extends EmailController {
+	// Defaults; will be overridden in subclasses
+	const TRACKING_CATEGORY_EN = TrackingCategories::DEFAULT_CATEGORY;
+	const TRACKING_CATEGORY_INT = TrackingCategories::DEFAULT_CATEGORY;
+
+	protected $wikiId;
+	protected $pageViews;
+
+	public function initEmail() {
+		$this->wikiId = $this->request->getVal( 'wikiId' );
+		$this->pageViews = $this->request->getVal( 'pageViews' );
+		$this->assertValidParams();
+
+		$this->pageViews = number_format( $this->pageViews );
+	}
+
+	protected function getSummary() {
+		return $this->getMessage( 'emailext-founder-digest-summary',
+			\WikiMap::getWikiName( $this->wikiId ) )->parse();
+	}
+
+	protected function assertValidParams() {
+		$this->assertValidWikiId();
+		$this->assertPageViewsSet();
+	}
+
+	/**
+	 * Asserts that the wikiId parameter is valid
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertValidWikiId() {
+		if ( $this->wikiId < 1 ) {
+			throw new Check( 'Invalid value passed for `wikiId`' );
+		}
+	}
+
+	/**
+	 * Asserts that the `pageViews` value is passed
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertPageViewsSet() {
+		if ( empty( $this->pageViews ) && $this->pageViews !== '0' ) {
+			throw new Check( 'Invalid value passed for `pageViews`' );
+		}
+	}
+
+	/**
+	 * Determine which sendgrid category to send based on target language and specific
+	 * founder email being sent. See dependent classes for overridden values
+	 *
+	 * @return string
+	 */
+	public function getSendGridCategory() {
+		return strtolower( $this->targetLang ) == 'en'
+			? static::TRACKING_CATEGORY_EN
+			: static::TRACKING_CATEGORY_INT;
+	}
+
+	protected static function getEmailSpecificFormFields() {
+		$formFields = [
+			'inputs' => [
+				[
+					'type' => 'text',
+					'name' => 'wikiId',
+					'label' => 'Wiki ID',
+					'tooltip' => 'ID number of wiki community'
+				],
+				[
+					'type' => 'text',
+					'name' => 'pageViews',
+					'label' => 'Page views',
+					'tooltip' => 'Number of page views'
+				]
+			]
+		];
+
+		return array_merge_recursive( $formFields, parent::getEmailSpecificFormFields() );
+	}
+}
+
+class FounderActivityDigestController extends FounderDigestController {
+	const TRACKING_CATEGORY_EN = TrackingCategories::FOUNDER_ACTIVITY_DIGEST_EN;
+	const TRACKING_CATEGORY_INT = TrackingCategories::FOUNDER_ACTIVITY_DIGEST_INT;
+
+	protected $pageEdits;
+	protected $newUsers;
+
+	public function initEmail() {
+		$this->pageEdits = $this->request->getVal( 'pageEdits' );
+		$this->newUsers = $this->request->getVal( 'newUsers' );
+		// Parent method is called here so that assertion of parameters is done at the right time
+		parent::initEmail();
+		$this->pageEdits = number_format( $this->pageEdits );
+		$this->newUsers = number_format( $this->newUsers );
```

Comment:
Same comment for number_format as above.  Maybe we should create a method in the parent class that returns the language object from targetLang so we don't have to construct it every time.